content/blog/2024-07-01-arrow-2-0.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Arrow 2.0 release
+category: articles
+tags: [core, articles]
+---
+
+# Arrow 2.0 release
+
+We are happy to announce the next major release of Arrow, version 2.0!
+As previously announced, migrating your projects to this release should be hassle-free
+if your code compiled in 1.2.x without any deprecation warnings
+(except for the breaking change in optics generation discussed below).
+
+This release is built with the new K2 compiler, and this gives us the ability
+to support a wider range of platforms, including WebAssembly. From now on, we shall
+provide artifacts for every platform supported by Kotlin.
+
+Apart from stabilization and general bug fixing, the theme of this release
+is improving the different DSLs provided by Arrow libraries. Our goal is to
+empower developers to write more succinct and readable code.
+
+## Simple accumulation in Raise
+
+TODO: Talk about `accumulating`
+
+Using Arrow Core data types as part of serialized data requires additional integration.
+In 1.2.x we started providing compile-time support for `kotlinx.serialization`. 
+From 2.0 we also provide `ArrowModule` for
+[contextual serialization](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md#contextual-serialization). This is needed, among others, when the data is processed
+by Ktor.
+
+## Additions to Fx
+
+Writing coroutine-heavy code may become cumbersome over time, especially if
+one intends to use as much concurrency as possible. Arrow Fx includes a `parZip`
+function, but not everybody enjoys having so many brackets.
+
+```kotlin
+parZip(
+    { downloadFile() },
+    { loadDataFromDatabase() }
+) { file, data -> Result(file, data) }
+```
+
+The new `awaitAll` scope tries to improve the situation by tweaking the
+usual `async` mechanism, ensuring that all `Deferred` values are `await`ed
+once the first one is requested. That means that the previous code behaves
+identically to the following.
+
+```kotlin
+awaitAll {
+    val file = async { downloadFile() }
+    val data = async { loadDataFromDatabase() }
+    Result(file.await(), data.await())
+}
+```
+
+We've also improved the STM block by allowing delegation as a means to
+read or change the value of a `TVar`.
+
+```kotlin
+fun STM.deposit(accVar: TVar<Int>, amount: Int): Unit {
+    val acc by accVar       // delegation here
+    val current = acc       // implicit 'read'
+    acc = current + amount  // implicit 'write'
+}
+```
+
+## Clearer retries for particular exceptions
+
+Until now, the `retry` operation in the Resilience module would capture
+any `Throwable` exception. From version 2.0 on you can specify a subclass
+of `Throwable` to be the target for retrying, whereas the rest of 
+exceptions will bubble as usual.
+
+```kotlin
+Schedule.recurs<Throwable>(2)
+        .retry<IllegalArgumentException, _> { ... }
+```
```

Comment:
See other comment, I think it is important we can call `doWhile` on `IllegalArgumentException` here as well.